crates/fuel-core/src/graphql_api/storage/da_compression.rs

```diff
@@ -0,0 +1,169 @@
+use fuel_core_compression::{
+    RegistryKeyspace,
+    RegistryKeyspaceValue,
+};
+use fuel_core_storage::{
+    blueprint::plain::Plain,
+    codec::{
+        postcard::Postcard,
+        primitive::Primitive,
+        raw::Raw,
+    },
+    structured_storage::TableWithBlueprint,
+    Mappable,
+};
+use fuel_core_types::{
+    fuel_compression::RegistryKey,
+    fuel_types::BlockHeight,
+};
+
+pub struct DaCompressedBlocks;
+
+impl Mappable for DaCompressedBlocks {
+    type Key = Self::OwnedKey;
+    type OwnedKey = BlockHeight;
+    type Value = Self::OwnedValue;
+    type OwnedValue = Vec<u8>;
+}
+
+impl TableWithBlueprint for DaCompressedBlocks {
+    type Blueprint = Plain<Primitive<4>, Raw>;
+    type Column = super::Column;
+
+    fn column() -> Self::Column {
+        Self::Column::DaCompressedBlocks
+    }
+}
+
+pub struct DaCompressionTemporalRegistry;
+
+impl Mappable for DaCompressionTemporalRegistry {
+    type Key = Self::OwnedKey;
+    type OwnedKey = (RegistryKeyspace, RegistryKey);
+    type Value = Self::OwnedValue;
+    type OwnedValue = RegistryKeyspaceValue;
+}
+
+impl TableWithBlueprint for DaCompressionTemporalRegistry {
+    type Blueprint = Plain<Postcard, Postcard>;
+    type Column = super::Column;
+
+    fn column() -> Self::Column {
+        Self::Column::DaCompressionTemporalRegistry
+    }
+}
+
+pub struct DaCompressionTemporalRegistryIndex;
+
+impl Mappable for DaCompressionTemporalRegistryIndex {
+    type Key = Self::OwnedKey;
+    // TODO: should we hash the key?
+    type OwnedKey = RegistryKeyspaceValue;
```

Comment:
We should use hash in the case of scripts and predicates. I think it is better to define a separate enum just for this table

Reply:
Moved to use a hash in 946a2e8f23fef2109fb9ba286ebd2d451c2af7cb. The enum is gone.